marshal.go

```diff
@@ -70,49 +71,99 @@ func Marshal(l *LDIF) (data string, err error) {
-				data += "delete: " + mod.Type + "\n"
-				for _, v := range mod.Vals {
-					ev, t := encodeValue(v)
-					col := ": "
-					if t {
-						col = ":: "
+					data += "-\n"
+				// delete operation - https://tools.ietf.org/html/rfc4511#section-4.6
+				case 1:
+					data += "delete: " + mod.Modification.Type + "\n"
+					for _, v := range mod.Modification.Vals {
+						ev, t := encodeValue(v)
+						col := ": "
+						if t {
+							col = ":: "
+						}
+						data += foldLine(mod.Modification.Type+col+ev, fw) + "\n"
 					}
-					data += foldLine(mod.Type+col+ev, fw) + "\n"
-				}
-				data += "-\n"
-			}
-			for _, mod := range e.Modify.ReplaceAttributes {
-				if len(mod.Vals) == 0 {
-					return "", errors.New("changetype 'modify', op 'replace' requires non empty value list")
-				}
-				data += "replace: " + mod.Type + "\n"
-				for _, v := range mod.Vals {
-					ev, t := encodeValue(v)
-					col := ": "
-					if t {
-						col = ":: "
+					data += "-\n"
+				// replace operation - https://tools.ietf.org/html/rfc4511#section-4.6
+				case 2:
+					if len(mod.Modification.Vals) == 0 {
+						return "", errors.New("changetype 'modify', op 'replace' requires non empty value list")
 					}
-					data += foldLine(mod.Type+col+ev, fw) + "\n"
+					data += "replace: " + mod.Modification.Type + "\n"
+					for _, v := range mod.Modification.Vals {
+						ev, t := encodeValue(v)
+						col := ": "
+						if t {
+							col = ":: "
+						}
+						data += foldLine(mod.Modification.Type+col+ev, fw) + "\n"
+					}
+					data += "-\n"
+				default:
+					return "", fmt.Errorf("invalid type %s in modify request", mod.Modification.Type)
 				}
-				data += "-\n"
 			}
+			// for _, mod := range e.Modify.AddAttributes {
+			// 	if len(mod.Vals) == 0 {
+			// 		return "", errors.New("changetype 'modify', op 'add' requires non empty value list")
+			// 	}
+
+			// 	data += "add: " + mod.Type + "\n"
+			// 	for _, v := range mod.Vals {
+			// 		ev, t := encodeValue(v)
+			// 		col := ": "
+			// 		if t {
+			// 			col = ":: "
+			// 		}
+			// 		data += foldLine(mod.Type+col+ev, fw) + "\n"
+			// 	}
+			// 	data += "-\n"
+			// }
+			// for _, mod := range e.Modify.DeleteAttributes {
+			// 	data += "delete: " + mod.Type + "\n"
+			// 	for _, v := range mod.Vals {
+			// 		ev, t := encodeValue(v)
+			// 		col := ": "
+			// 		if t {
+			// 			col = ":: "
+			// 		}
+			// 		data += foldLine(mod.Type+col+ev, fw) + "\n"
+			// 	}
+			// 	data += "-\n"
+			// }
+			// for _, mod := range e.Modify.ReplaceAttributes {
+			// 	if len(mod.Vals) == 0 {
+			// 		return "", errors.New("changetype 'modify', op 'replace' requires non empty value list")
+			// 	}
+			// 	data += "replace: " + mod.Type + "\n"
+			// 	for _, v := range mod.Vals {
+			// 		ev, t := encodeValue(v)
+			// 		col := ": "
+			// 		if t {
+			// 			col = ":: "
+			// 		}
+			// 		data += foldLine(mod.Type+col+ev, fw) + "\n"
+			// 	}
+			// 	data += "-\n"
+			// }
```

Comment:
Do we need to keep these commented lines?

Reply:
Deleted